Orange/widgets/utils/itemmodels.py

```diff
@@ -903,6 +904,8 @@ def set_domain(self, domain):
                     *(vars for i, vars in enumerate(
                         (domain.attributes, domain.class_vars, domain.metas))
                       if (1 << i) & section)))
+                if self.skip_hidden_vars:
+                    to_add = filter_visible(to_add)
```

Comment:
`to_add = list(filter_visible(to_add))`

This is needed because of `if to_add` [later on](https://github.com/biolab/orange3/pull/2690/files#diff-4e5076ff2a69272284062b73fbddf0faR918), which tests whether there is some more content and a separator needs to be added. Since `filter_visible` returns a generator, the condition is always `True` and separator is added even when the generator would return nothing.

Reply:
It should be fixed now. Though, in the future, we might consider changing `filter_visible` to return a tuple instead of a generator.